lib/textstat.rb

```diff
@@ -161,58 +155,49 @@ def self.difficult_words(text, language = 'en_us')
     text_list = text.downcase.gsub(/[^0-9a-z ]/i, '').split(' ')
     diff_words_set = Set.new
     text_list.each do |value|
-      unless easy_words.include? value
-        if syllable_count(value) > 1
-          diff_words_set.add(value)
-        end
-      end
+      next if easy_words.include? value
+
+      diff_words_set.add(value) if syllable_count(value) > 1
     end
-    return diff_words_set.length
+    diff_words_set.length
   end
 
   def self.dale_chall_readability_score(text)
     word_count = lexicon_count(text)
     count = word_count - difficult_words(text)
 
     begin
-      per = count.to_f / word_count.to_f * 100
+      per = 100.0 * count / word_count
     rescue ZeroDivisionError
       return 0.0
     end
 
     difficult_words = 100 - per
-    score = (
-    (0.1579 * difficult_words)
-    + (0.0496 * avg_sentence_length(text)))
+    score = 0.1579 * difficult_words + 0.0496 * avg_sentence_length(text)
+    score += 3.6365 if difficult_words > 5
 
-    if difficult_words > 5
-      score += 3.6365
-    end
-    return score.round(2)
+    score.round(2)
   end
 
   def self.gunning_fog(text)
-    begin
-      per_diff_words = (
-      (difficult_words(text) / lexicon_count(text) * 100) + 5)
+    per_diff_words = 100.0 * difficult_words(text) / lexicon_count(text) + 5
```

Comment:
can you explain where the `+ 5` at the end comes from? From what I've read about Gunning Fog it should just be `100.0 * difficult_words(text) / lexicon_count(text)`

Reply:
Also, you'll note I put the `100.0` at the start, thus resulting in the difficult_words / lexicon_count being floating point math (and as a result a significant change in the spec test. Per below, I couldn't find anything to suggest that this should have been integer math???